build.gradle

```diff
@@ -76,4 +76,132 @@ project(':configurator') {
                     'Main-Class': mainClassName)
         }
     }
-}
\ No newline at end of file
+}
+ext {
+    osNames = ['linux', 'windows', 'mac']
+    jreFileNameTemplate = {sys,extension -> "OpenJDK8U-jre_x64_${sys}_hotspot_8u212b03.${extension}" }
+    jreDownloadDir = file("$buildDir/jre-downloads")
+    jreExtractDir = "${jreDownloadDir.absolutePath}/jre"
+    jreExtractSubDir = {sys -> "${jreExtractDir}/$sys"}
+    jreHashes = ['74daf0b77a7fd679cbb3a6228e0efa8c4a90b7664aa057f211e34bbfb38640fb',
+                 'c09bab89cd82483c371597c5c364094a145c1fbba43a1d3d7c3e350b89dedc89',
+                 '80855320c42a3b06617c6e466c64df67731542a805972185b075ca6cb1222c7f']
+    packageFileExtension = ['zip', 'tar.gz']
+}
+
+import java.security.MessageDigest
+
+task downloadJre(description: 'Download OpenJRE of Linux, Windows and Mac, and check their hashes') {
+
+    doFirst {
+        if(!jreDownloadDir.exists()) {
+            jreDownloadDir.mkdirs()
+        }
+        osNames.each {String sys ->
+            def index = osNames.indexOf(sys)
+            def jreFileExtension = (sys == 'windows' ? packageFileExtension[0] : packageFileExtension[1])
+            def jreFileName = jreFileNameTemplate(sys,jreFileExtension)
+            def jreFile = file("$jreDownloadDir.absolutePath/$jreFileName")
+            if(!jreFile.exists()){
+                new URL("https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u212-b03/$jreFileName")
+                        .withInputStream {
+                    jreFile.newOutputStream() << it
+                }
+                final hash = MessageDigest.getInstance('SHA-256').with {
+                    jreFile.eachByte(1024) {buff,len ->
+                        update(buff, 0, len)
+                    }
+                    digest().encodeHex() as String
+                }
```

Comment:
This is just a thought (i.e. no action required), but we could actually calculate the hash at download time in the above closure... But hey. ;)